bugfixtemplates/anymatch-using-findfirst/CheckAnyMatchUsingFindFirst.java

```diff
@@ -0,0 +1,31 @@
+package bugfixtemplates;
+
+import java.util.Collection;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+class CheckAnyMatchUsingFindFirst {
+    
+    Collection<String> col;
+
+    void positiveFilterThenFindFirstAndOnlyCheckPresent() {
+        col.stream().filter(null).findFirst().isPresent();
+    }
+
+    void negativeFilterThenFindFirstButAlsoNeedsTheMatch() {
+        Optional<String> any = col.stream().filter(null).findFirst();
+        if (any.isPresent())
+            any.get();
+    }
+
+    void positiveFilterThenFindAnyAndOnlyCheckPresent() {
+        col.stream().filter(null).findAny().isPresent();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It is more convenient and readable to directly use Stream::anyMatch instead of Stream::filter, Stream::findAny, and Optional::isPresent when the matched data returned by Stream::findAny is not used anywhere else.